### PR TITLE
Add poll blocking context

### PR DESCRIPTION
**Pull request by slacrherbst**
This PR adds a poller blocking context to temporarily pause polling while the context is active:

with root.pollBlock():
     --- do some stuff

### CODE REVIEW

File: python/pyrogue/_PollQueue.py [hunk 1]
```diff
@@ -41,11 +41,11 @@ def __init__(self,*, root):
         self._pq = [] # The heap queue
         self._entries = {} # {Block: Entry} mapping to look up if a block is already in the queue
         self._counter = itertools.count()
-        self._lock = threading.RLock()
-        self._update = threading.Condition()
+        self._clock = threading.Condition(threading.RLock())
```
Comment (bengineerd) [inline python/pyrogue/_PollQueue.py]: Maybe name this `self._cond`? It took me a sec the realize that `clock` is short for Condition Lock.

File: python/pyrogue/_PollQueue.py [hunk 2]
```diff
@@ -41,11 +41,11 @@ def __init__(self,*, root):
...
         self._run = True
         self._pause = True
         self._root = root
+        self._bcount = 0
```
Comment (bengineerd) [inline python/pyrogue/_PollQueue.py]: Also just name this `_blockCount`.

File: python/pyrogue/_PollQueue.py (hunk 1 above)
Comment (slacrherbst) [reply python/pyrogue/_PollQueue.py]: Yeah I can rename. The main point is that you can use one object for all locking here. maybe condLock?

File: python/pyrogue/_PollQueue.py (hunk 2 above)
Comment (slacrherbst) [reply python/pyrogue/_PollQueue.py]: Ok. will do.

**Comment by slacrherbst** [comment]
Ok so the global lock you are thinking of would be for extreme cases where you want to block all other entry points (gui, epcis, zmq, etc, poll) while the operation occurs? So the blocking thread would do:

with root.exclusive():
    do some stuff

and all of the other entry points would need to do:

with root.nonexclusive():
    do some stuff.

**Comment by slacrherbst** [comment]
We would need to be careful where we place the:

with root.nonexclusive():

call since it could kill performance if done wrong. But maybe we start with this model for poll blocking and we can expand upon it with careful testing.

**Comment by slacrherbst** [comment]
I think we should stick with the poll block for now. Since some of the clients access variables directly it is hard to set a good entry point for other locks.

**Pull request merged**